saltstack/base/salt/prerequisites/usr/bin/user-data-helper.sh

```diff
@@ -82,13 +82,13 @@ create_luks_volume() {
   echo "$SECRET_ENCRYPTION_KEY_SOURCE" > "$ENCRYPTION_KEY_FILE"
 
   echo "LUKS volume creation started."
-  /etc/cdp-luks/bin/create-luks-volume.sh
+  /etc/cdp-luks/bin/create-luks-volume.sh | tee /var/log/cdp-luks/create-luks-volume.log
   echo "LUKS volume creation finished."
 }
 
 populate_luks_volume() {
   echo "LUKS volume population with secrets started."
-  /etc/cdp-luks/bin/populate-luks-volume.sh
+  /etc/cdp-luks/bin/populate-luks-volume.sh | tee /var/log/cdp-luks/populate-luks-volume.log
   echo "LUKS volume population with secrets finished."
```

Comment:
Sorry, forgot mentioning that we should think about the `${PIPESTATUS[0]}` thing in these script executions as well. There are two choices: either `exit` explicitly abort if `${PIPESTATUS[0]}` is nonzero, or just `return` from the function with the nonzero `${PIPESTATUS[0]}`. As I see, https://github.infra.cloudera.com/cloudbreak/cloudbreak/pull/2733 / [CB-27096](https://jira.cloudera.com/browse/CB-27096) will force `user-data-helper.sh` abort if any command execution fails, so even the `return` option is satisfactory.